alphametics/example.go

```diff
@@ -0,0 +1,200 @@
+package alphametics
+
+import "sync"
+
+const TestVersion = 1
+
+// Set this to true to debug the example. It enables several useful logging tools.
+const debugging = false
+
+type accumulator struct {
+	carry    int
+	usedNums uint16       // bit n denotes if digit n is taken
+	solution map[rune]int // partially filled solution map
+}
+
+type ipc struct {
+	resultChan chan map[rune]int
+	abortChan  chan struct{}
+	waitGroup  *sync.WaitGroup
+}
+
+func Solve(a, b, c string) map[rune]int {
+	aRunes, bRunes, cRunes := prepareRunes(a, b, c)
+
+	// Setup the machinery for solving in parallel.
+	resultChan := make(chan map[rune]int)
```

Comment:
The concurrency here is pretty confusing.
You might try handling the waiting here in the caller rather than giving the waitgroup to the step solver.

So something like
```go

go func() {
    solveStep(...)
    waitGroup.Done()
}()
```

Or alternatively handle the concurrency at a lower level may help clean this up a bit as I see you are also adding to the waitgroup in the `solveStep` fn, which is recursive.

Cancellation may also be better served by a `context.WithCancel`.